src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp

```diff
@@ -472,9 +472,10 @@ namespace OpenMS
     }
 
     // add back peptide identification id if it got skipped (decoys)
-    for (auto p : peptides)
+    for (const auto& pep : peptides)
```

Comment:
You can also flag the copy-for-loop check. Not sure if it makes a difference here now though.